### PR TITLE
Add scalar and vector testing for station point interp

### PR DESCRIPTION
**Pull request by AlexanderRichert-NOAA**
This PR adds CI testing for station point interpolation. These changes were made under the grib2-based testing. Output data values are based on real runs and have not been independently validated.

### CODE REVIEW

File: tests/interp_mod_grib2.F90 [hunk 1]
```diff
@@ -64,9 +64,11 @@ subroutine interp(grid, interp_opt)
```
Comment (edwardhartnett) [inline tests/interp_mod_grib2.F90:64]: Can you please name all NCEPLIBS tests "test_SOMETHING.F90" for Fortran and "tst_SOMETHING.c" for C.

So all Fortran test programs should have a name "test_*.F90".

**Comment by edwardhartnett** [comment]
OK, actually we have 62% test coverage right now, which is much better than I thought, so that's great. 

We do have several library functions with 0 testing however...

**Review by edwardhartnett — APPROVED** [review]

**Comment by AlexanderRichert-NOAA** [comment]
@edwardhartnett I'm seeing differences between the gridded and station points output, i.e., if I take some of the outputs/coordinate values from the gridded scheme and calculate those locations as station points, I get difference data values. Is this expected based on grid/cell geometry?

**Comment by AlexanderRichert-NOAA** [comment]
Nevermind, I was handling my coordinates wrong. Looks like we're okay.

**Pull request merged**